client/post-editor/controller.js

```diff
@@ -91,13 +92,21 @@ function maybeRedirect( context, postType, site ) {
 	return false;
 }
 
-function getPressThisContent( text, url, title ) {
-	return ReactDomServer.renderToStaticMarkup(
-		<p>
-			{ text ? <blockquote>{ text }</blockquote> : null }
-			via <a href={ url }>{ title }</a>.
-		</p>
-	);
+function getPressThisContent( text, url, title, image, embed ) {
+	const pieces = [];
+	if ( image ) {
+		pieces.push( ReactDomServer.renderToStaticMarkup( <p><a href={ url }><img src={ image } /></a></p> ) );
+	}
+	if ( isValidUrl( embed ) ) {
```

Comment:
>`<script>alert('hi')</script>`

Maybe we should leverage `ReactDomServer` then, with added bonus of consistency?

```jsx
pieces.push( ReactDomServer.renderToStaticMarkup( <p>{ embed }</p> ) );
```